### PR TITLE
[v2] Add smart default context-menu functionality (backported from v3)

### PR DESCRIPTION
**Pull request by mmghv**
# Description

This PR backports the "smart" default context-menu functionality from v3.

This feature allows the default context-menu to only be available in text contexts by default, which can be controlled with the `--default-contextmenu` CSS style, this feature only works in production, in dev & debug the full default context-menu is always enabled.

Reference #2733 

## Type of change
  
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?
  
- [x] Windows
- [ ] macOS
- [x] Linux

### How to test :

Enable the `EnableDefaultContextMenu` app option, then using this [index.zip](https://github.com/wailsapp/wails/files/11881866/index.zip) html file, you can test all scenarios, in production it should work, in dev & debug it should have no effect and the default context-menu should always be available.

# Checklist:

- [x] I have updated `website/src/pages/changelog.mdx` with details of this PR
- [x] My code follows the general coding style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: website/docs/reference/options.mdx [hunk 1]
```diff
@@ -420,8 +420,28 @@ Type: `string`
...
+| CSS Style                      | Behavior                                                                                                                                           |
+|--------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------|
+| `--default-contextmenu: auto;` | (**default**) will show the default context menu only if :<br/> contentEditable is true OR text has been selected OR element is input or textarea  |
+| `--default-contextmenu: show;` | will always show the default context menu                                                                                                          |
+| `--default-contextmenu: hide;` | will always hide the default context menu                                                                                                          |
+
+This rule is inherited like any normal CSS rule, so nesting works as expected.
+
+:::note
+This filtering functionality is only enabled in production, so in development and in debug build, the full context-menu is always available everywhere.
```
Comment (leaanthony) [inline website/docs/reference/options.mdx:438]: I think that if you right click a background with no text in debug mode, it won't be available as the current check is CSS only - it has no idea it's in dev mode. We could implement that by setting `flags.debug` to `true` and checking it as part of `processDefaultContextMenu`

**Review by leaanthony — CHANGES_REQUESTED** [review]

File: website/docs/reference/options.mdx (hunk 1 above)
Comment (mmghv) [reply website/docs/reference/options.mdx:438]: It skips it all together in the runtime in dev & debug, so right clicking anywhere shows the context menu in dev & debug, that's why I needed the previous PR 😉 

https://github.com/wailsapp/wails/pull/2748/files#diff-110ff4bc4cc00d585c83f35398997ba75ba4dca549a87847cf50cb37a62ac240R190-R199

You could test locally to confirm 👍

Comment (leaanthony) [reply website/docs/reference/options.mdx:438]: Ooooh! I totally missed that nuance! Awesome.

**Review by leaanthony — APPROVED** [review]
Thanks for doing this 🙏

**Pull request merged**